### PR TITLE
Use new figma design for query builder

### LINKED ISSUE #4093: Filter usability is incredibly frustrating
Two example of filtering being quite frustrating for our customers.

1) "Filter usability is incredibly frustrating. "

For example, when entering more than one parameter to filter by (ex: URL contains X, Y, ...), editing from already entered data (ex: editing an existing URL query that you want to change because you made a typo), etc. -> this also is annoying because you can't use the arrow keys to highlight the list and press space to select/unselect. But more importantly, when you enter a new item and press enter it just closes the filter and doesn't actually add the item.

Link to discord: https://discord.com/channels/1026884757667188757/1052834249868324895/1068966214682018023

2) Here is my other problem, I face, so when I select filter types, I hit this bug where I can't click on things like "is between" or "is not between", I have to type it then click my enter key

https://user-images.githubusercontent.com/20292680/217410335-8f66d040-5f54-4449-a567-7abd02526784.mov

Link to discord: https://discord.com/channels/1026884757667188757/1052834219522523178/1072698410106372136

### LINKED ISSUE #5040: implement filter dropdown Errors/Sessions using figma design
https://www.figma.com/file/pAHlb8IFmVSewPJcphbjIH/Component-Library?node-id=424-8811&t=roqpiRvZkabjR2kN-11

@julian-highlight is there a new style for dropdowns? any other design improvements we should make here?

### PR DESCRIPTION
**Pull request by mayberryzane**
## Summary
- Implements the design here: https://www.figma.com/file/rdBPNzn7Klk6RG1LHUCE5B/Screen-Designs?node-id=5020%3A222338&mode=dev
- Creates a new `ComboboxSelect` component loosely following the ariakit select w/ combobox example: https://ariakit.org/examples/select-combobox
- closes #5040
- closes #4093

## How did you test this change?
- clicktested locally

## Are there any deployment considerations?
- no, can revert if issues

### CODE REVIEW

**Review by Vadman97 — COMMENTED** [review]
some bugs i found in clicktesting
https://www.loom.com/share/c499f85bc3064c8bbfe619a82375b769

**Comment by mayberryzane** [comment]
@Vadman97 - thanks for the feedback! fixed all of those items and a few other bugs, looking into a little stuttering but should be good to review now - also just wanted to note some slight style differences between the preview env and local:

preview:
Screen Shot 2023-09-07 at 8 28 48 AM
local:
Screen Shot 2023-09-07 at 8 28 41 AM

**Review by Vadman97 — APPROVED** [review]
yup, looks great now! the css styling weirdness is just an issue in reflame.

File: frontend/src/components/QueryBuilder/QueryBuilder.tsx [hunk 1]
```diff
@@ -2036,69 +1809,48 @@ function QueryBuilder(props: QueryBuilderProps) {
...
+							? [
+									<Tag
+										shape="basic"
+										kind="secondary"
+										emphasis="low"
+										onClick={toggleIsAnd}
+										key={`separator-${index}`}
+										disabled={readonly}
+									>
+										{isAnd ? 'and' : 'or'}
```
Comment (Vadman97) [inline frontend/src/components/QueryBuilder/QueryBuilder.tsx:1833]: @mayberryzane i think there is a new bug that causes results to not update when switching `or/and`
https://www.loom.com/share/daac0ac7f5a04c4fbc168cccf5286835

Comment (mayberryzane) [reply frontend/src/components/QueryBuilder/QueryBuilder.tsx:1833]: same bug exists in prod rn but it's probably an easy fix, taking a look

Comment (Vadman97) [reply frontend/src/components/QueryBuilder/QueryBuilder.tsx:1833]: sg, if u'd prefer to fix in separate pr, feel free to ship this

**Pull request merged**